src/vanilla/Model/CompositeTypeTS.cs

```diff
@@ -67,6 +67,38 @@ public IReadOnlyList<CompositeTypeTS> ImmediatePolymorphicSubtypes
 
         public string NameAsFileName => Name.EqualsIgnoreCase("index") ? "IndexModelType" : (string)Name;
 
+        public IModelType AdditionalProperties { get; set; }
+
+        public string AdditionalPropertiesTSType()
+        {
+            string result = "any";
+            if (AdditionalProperties != null)
+            {
+                var type = AdditionalProperties.TSType(true);
+                result = type != "any" ? $"{type} | any" : "any";
+            }
+            return result;
+        }
+
+        public string AdditionalPropertiesDocumentation()
+        {
+            string result = "Describes unknown properties. ";
+            if (AdditionalProperties != null)
+            {
+                var type = AdditionalProperties.TSType(true);
+                if (type != "any")
+                {
+                    result += $"The value of an unknown property MUST be of type \"{type}\". Due to valid TS constraints " +
+                        $"we have modeled this as a union of `{type} | any`.";
+                }
+                else
+                {
+                    result += "The value of an unknown property can be of \"any\" type.";
+                }
```

Comment:
Is it possible for AdditionalProperties to have a description property specified? If so, then I'd add that description first.

Reply:
The AdditionalProperties property is created by the modeler. It has the generated description "Unmatched properties from the message are deserialized this collection".

It doesn't appear to be possible to add a description for the `"additionalProperties"` in Swagger, so I think I'm content with this approach.